rasa/shared/core/domain.py

```diff
@@ -239,6 +239,11 @@ def from_dict(cls, data: Dict) -> "Domain":
         if domain_slots:
             rasa.shared.core.slot_mappings.validate_slot_mappings(domain_slots)
         slots = cls.collect_slots(domain_slots)
+        domain_actions = data.get(KEY_ACTIONS, [])
+        actions = cls._collect_actions(domain_actions)
+        actions_which_explicitly_need_domain = (
+            cls._collect_actions_which_explicitly_need_domain(domain_actions)
+        )
```

Comment:
I'd rather recommend moving this logic into the constructor, rather than modifying the signature of the constructor (it's already super long 😅 )